sklearn/metrics/_regression.py

```diff
@@ -194,6 +195,85 @@ def mean_absolute_error(y_true, y_pred, *,
     return np.average(output_errors, weights=multioutput)
 
 
+def pinball_loss(y_true, y_pred, *,
+                 sample_weight=None,
+                 alpha=0.5,
+                 multioutput='uniform_average'):
+    """Pinball loss for quantile regression.
+
+    Read more in the :ref:`User Guide <pinball_loss>`.
+
+    Parameters
+    ----------
+    y_true : array-like of shape (n_samples,) or (n_samples, n_outputs)
+        Ground truth (correct) target values.
+
+    y_pred : array-like of shape (n_samples,) or (n_samples, n_outputs)
+        Estimated target values.
+
+    sample_weight : array-like of shape (n_samples,), default=None
+        Sample weights.
+
+    alpha: double, slope of the pinball loss, default=0.5,
+        this loss is equivalent to :ref:`mean_absolute_error` when `alpha=0.5`,
+        `alpha=0.95` is minimized by estimators of the 95th percentile.
+
+    multioutput : {'raw_values', 'uniform_average'}  or array-like of shape \
+            (n_outputs,), default='uniform_average'
+        Defines aggregating of multiple output values.
+        Array-like value defines weights used to average errors.
+
+        'raw_values' :
+            Returns a full set of errors in case of multioutput input.
+
+        'uniform_average' :
+            Errors of all outputs are averaged with uniform weight.
+
+
+    Returns
+    -------
+    loss : float or ndarray of floats
+        If multioutput is 'raw_values', then mean absolute error is returned
+        for each output separately.
+        If multioutput is 'uniform_average' or an ndarray of weights, then the
+        weighted average of all output errors is returned.
+
+        The pinball loss output is a non-negative floating point. The best
+        value is 0.0.
+
+    Examples
+    --------
+    >>> from sklearn.metrics import pinball_loss
+    >>> y_true = [3, -0.5, 2, 7]
+    >>> y_pred = [2.5, 0.0, 2, 8]
+    >>> pinball_loss(y_true, y_pred)
+    0.25
+    >>> y_true = [3, -0.5, 2, 7]
+    >>> y_pred = [2.5, 0.0, 2, 8]
+    >>> pinball_loss(y_true, y_pred, alpha=0.1)
+    0.35
+    """
+    y_type, y_true, y_pred, multioutput = _check_reg_targets(
+        y_true, y_pred, multioutput)
+    check_consistent_length(y_true, y_pred, sample_weight)
+    diff = y_pred - y_true
+    sign = (diff >= 0).astype(diff.dtype)
+    loss = diff * sign * (1 - alpha) - diff * (1 - sign) * alpha
+    output_errors = np.average(loss, weights=sample_weight, axis=0)
```

Comment:
@sdpython I think @lorentzenchr suggests the name `mean_pinball_loss` because of the `np.average` across samples here: same as for mean squared error and mean absolute error.

Reply:
But as I said earlier, I don't think it's necessary to be that explicit.